### PR TITLE
Query History: Implement RemoteStorage methods: get all, add new 

### LINKED ISSUE #47252: Query History: Implement RemoteStorage methods: get all, add new
Use get all and add new endpoints to create query history.

Also ensure that the query history is not reloaded when the container is not opened in public/app/features/explore/state/query.ts, handleHistory() method

### PR DESCRIPTION
**Pull request by ifrost**
**What this PR does / why we need it**:

First implementation of the remote storage API supporting:

*  adding a new query to the history (after running it)
* getting the list of items in query history:
  * input search and data source filtering 
  * filtering by starred items cannot be tested yet as the API to star an item is not added yet (to be done in #48406)
  * limited to 100 results (pagination/load more will be implemented in #48407)
  * settings are hardcoded, to be implemented in #48408

**Which issue(s) this PR fixes**:

Fixes #47252

**Special notes for your reviewer**:

How to test it?

Enable query history:

```
[query_history]
enabled = true
```

Once query history is enabled it doesn't use local storage anymore so you start with a clean slate. Migration from local storage will be implemented in #48409.

Run a few queries and check if they are available in query history. Filtering by data source and search input should work as expected. Filtering by time, starring, commenting, deleting items, and changing settings is not supported yet.

### CODE REVIEW

File: public/app/core/history/RichHistoryStorage.ts [hunk 1]
```diff
@@ -50,3 +51,19 @@ export default interface RichHistoryStorage {
   getSettings(): Promise<RichHistorySettings>;
   updateSettings(settings: RichHistorySettings): Promise<void>;
 }
+
+/**
+ * Simple data source srv mock used for testing remote and local storage conversions between
+ * uid and data source name.
+ */
+export const DataSourceSrvMock = {
```
Comment (Elfo404) [inline public/app/core/history/RichHistoryStorage.ts:59]: is this the right place for this? 🤔

Comment (ifrost) [reply public/app/core/history/RichHistoryStorage.ts]: Good point. Actually it's not really needed, might be easier/better to test it with real DataSourceSrv - 77e2336482ac06c5a63e4923b80dfb50d0153932

**Review by gelicia — APPROVED** [review]
I tested this by turning on the feature, running some queries and seeing them show up in the pane, and checking the sqlite db to ensure they are showing up there as well. Looks really good!

File: public/app/features/explore/RichHistory/RichHistoryStarredTab.tsx [hunk 2]
```diff
@@ -135,26 +138,30 @@ export function RichHistoryStarredTab(props: Props) {
           </div>
           <div aria-label="Sort queries" className={styles.sort}>
             <Select
-              value={sortOrderOptions.filter((order) => order.value === richHistorySearchFilters.sortOrder)}
-              options={sortOrderOptions}
+              value={getSortOrderOptions().filter((order) => order.value === richHistorySearchFilters.sortOrder)}
```
[the remark this answers is not in the document]
Comment (ifrost) [reply public/app/features/explore/RichHistory/RichHistoryStarredTab.tsx]: Fixed in ea696f8d4bcad835eef1936845985f08a7123d13 + 93be0d5b1b5e7ae5d1809241be27ad3586d86e52

**Pull request merged**